website/docs/developer/research/research-lambda-container.md

```diff
@@ -0,0 +1,139 @@
+---
+id: research-lambda-container
+title: Using Lambda functions as container research Notes
+description: Research Notes on 
+---
+
+## Overview
+
+Lambda functions can now be deployed as container images using Docker instead of zip files. This research webpage discusses how lambda functions can be prototyped as container and its pros and cons.
+
+## Pros and Cons of using container for lambdas
+
+### Pros
+- Defining runtime environment in a container image gives developers more control over their environment compared to what they get with predefined runtimes and zipping dependencies. 
+- you can now package and deploy Lambda functions as container images of up to 10 GB in size compared to only 250MB in case of lambda deployment package size. 
+- preferable for data-heavy or dependency-heavy applications.
+- The Lambda service provides a variety of base image options with pre-installed runtimes that will be patched and maintained by AWS.
+- There is no additional cost to use the container feature but you have to pay for the ECR repository.
+
+### Cons
+- Container support requires your Lambda function code point to an ECR Repo URI which means that repo also has to be maintained. This also includes additional cost to use the ECR service.
+- need additional work to create the Dockerfile, tag and push the image to ECR repo.
+- Need additional work for deleting older images under the ECR repository.
+
+:::warning
+Currently NGAP does not support using ECR repository to store images which could bring possible risk and challenges to using containers for lambdas.
+:::
+
+## New configuration for lambda container
+There are a few configuration that needs to be added if the lambda is deployed from docker image.
+- Architecture- This  determines the type of computer processor that Lambda uses to run the function. Lambda provides a choice of instruction set architectures which is either `arm64` or `x86_64`. The `arm64` architecture offer lower cost per Gb/s compared to the other one. Check this [link](https://docs.aws.amazon.com/lambda/latest/dg/foundation-arch.html?icmpid=docs_lambda_help) for additional information.
+- Image configuration- These are values that can be used to override the container image settings for `ENTRYPOINT`, `CMD`, `WORKDIR` and `ENV`.
+
+```
+  ENTRYPOINT – Specifies the absolute path of the entry point to the application.
+
+  CMD – Specifies parameters that you want to pass in with ENTRYPOINT.
+
+  WORKDIR – Specifies the absolute path of the working directory.
+
+  ENV – Specifies an environment variable for the Lambda function.
+```
+- Image URI- The location of the container image to use for your function.
+
+
+## Creating a prototype
+Creating or updating the function is done by building a Docker image, uploading the new version to ECR and deploying/updating the Lambda function to point to the newly uploaded image using terraform. Docker CLI has been used to perform the task.
```

Comment:
"the task" feels a bit vague. There are three distinct tasks in "creating or updating the function is done by building a Docker image, uploading the new version to ECR and deploying/updating the Lambda function to point to the newly uploaded image using terraform".

Reply:
yes reworded to 

`Docker CLI has been used to build, tag and push the container image to ECR`